tests/test_memory.py

```diff
@@ -0,0 +1,31 @@
+import asyncio
+import ssl
+
+import pytest
+import urllib3
+
+import truststore
+
+from .conftest import Server
+
+try:
+    import pytest_memray  # noqa: F401
+
+    memray_installed = True
+except ImportError:
+    memray_installed = False
+
+
+@pytest.mark.skipif(not memray_installed, reason="Memray isn't installed")
+@pytest.mark.limit_memory("1MB")
```

Comment:
This is worth something, but for catching leaks what we care about is not the high water mark but how much RAM is still used after all tests are done...right?

Reply:
I agree, I was trying to figure out how we can assert that no allocations were leaked but wasn't able to immediately find anything, I see [this PR](https://github.com/bloomberg/pytest-memray/pull/52) which looks like what we need? Maybe I'll add a ping of @pablogsal in case there's something I'm missing from the docs.

Reply:
@sethmlarson Okay yeah, that looks like the right place for it. We can follow up and add it once that PR lands.

Reply:
We are still iterating a bit over thar PR because is quite difficult to isolate from weird one-off allocations that the interpreter does and caches. Stay tuned. Meanwhile you can use the high watermark limit + a lot of iterations of the test as a proxy if you want